recipes-core/searcher-ssh-key/files/shell-wrapper.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+# Set proper environment
+export USER=searcher
+export LOGNAME=searcher
+export HOME=/home/searcher
+export SHELL=/bin/sh
+export PS1='searcher@tdx:\w\$ '
+
+# Check current mode and handle transition if needed
+mode=$(mode-controller status | awk '{print $NF}')
```

Comment:
An alternative would be to not read the mode here, instead call `mode-controller` with no arguments and let it figure out the mode.
I think `/usr/bin/ns-enter` can be called regardless of mode which is super convenient